pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -38,29 +49,231 @@ type Service struct {
 	store             storage.Store
 }
 
+//state defines a didcomm service state
+type state struct {
+	MsgType string            `json:"@type"`
+	ID      string            `json:"@id"`
```

Comment:
I don't think we need this. The thread's ID is enough (for now).

Reply:
we need that for invitation .. where thread id is not part of the request sent

Reply:
removed